spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -80,6 +99,22 @@ class ClientSession(val sessionId: Int, createRequest: CreateClientRequest)
     new JobStatus(id, status.state, status.result, status.error)
   }
 
+
+  private def copyResourceToHDFS(dataStream: InputStream, name: String): URI = {
+    val filePath = new Path(sessionHome, name)
+    val outFile = fs.create(filePath, true)
+    val buffer = new Array[Byte](512 * 1024)
+    var read = -1
+    try {
+      while ({read = dataStream.read(buffer); read != -1}) {
```

Comment:
Couldn't find one - Guava has a [ByteStreams#copy](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/ByteStreams.html#copy(java.io.InputStream,%20java.io.OutputStream)), but that uses a buffer too.